src/sdl/utils.cpp

```diff
@@ -395,86 +395,59 @@ surface scale_surface_legacy(const surface &surf, int w, int h)
-		PLAIN_LOG << "Create an empty image";
-		return dst;
-	}
-
-	if(surf == nullptr || dst == nullptr) {
+	surface dst(w, h);
+	if(dst == nullptr) {
 		PLAIN_LOG << "Could not create surface to scale onto";
 		return nullptr;
 	}
 
+	if(w == 0 || h == 0) {
+		PLAIN_LOG << "Creating an empty image";
+		return dst;
+	}
+
 	{
 		const_surface_lock src_lock(surf);
 		surface_lock dst_lock(dst);
 
 		const uint32_t* const src_pixels = src_lock.pixels();
 		uint32_t* const dst_pixels = dst_lock.pixels();
 
-		float xratio = static_cast<float>(surf->w) / w;
-		float yratio = static_cast<float>(surf->h) / h;
-
-		float ysrc = 0.0f;
-		for(int ydst = 0; ydst != h; ++ydst, ysrc += yratio) {
-			float xsrc = 0.0f;
-			for(int xdst = 0; xdst != w; ++xdst, xsrc += xratio) {
-				float red = 0.0f, green = 0.0f, blue = 0.0f, alpha = 0.0f;
-
-				float summation = 0.0f;
-
-				// We now have a rectangle, (xsrc,ysrc,xratio,yratio)
-				// which we want to derive the pixel from
-				for(float xloc = xsrc; xloc < xsrc+xratio; xloc += 1) {
-					const float xsize = std::min<float>(std::floor(xloc+1)-xloc,xsrc+xratio-xloc);
-
-					for(float yloc = ysrc; yloc < ysrc+yratio; yloc += 1) {
-						const int xsrcint = std::max<int>(0,std::min<int>(surf->w-1,static_cast<int>(xsrc)));
-						const int ysrcint = std::max<int>(0,std::min<int>(surf->h-1,static_cast<int>(ysrc)));
-						const float ysize = std::min<float>(std::floor(yloc+1)-yloc,ysrc+yratio-yloc);
-
-						uint8_t r,g,b,a;
-
-						SDL_GetRGBA(src_pixels[ysrcint*surf->w + xsrcint],surf->format,&r,&g,&b,&a);
-						float value = xsize * ysize;
-						summation += value;
-						if (!a) continue;
-						value *= a;
-						alpha += value;
-						red += r * value;
-						green += g * value;
-						blue += b * value;
-					}
-				}
-
-				if (alpha != 0) {
-					red = red / alpha + 0.5f;
-					green = green / alpha + 0.5f;
-					blue = blue / alpha + 0.5f;
-					alpha = alpha / summation + 0.5f;
-				}
-
-				dst_pixels[ydst*dst->w + xdst] = SDL_MapRGBA(
-				dst->format
-				, static_cast<uint8_t>(red)
-				, static_cast<uint8_t>(green)
-				, static_cast<uint8_t>(blue)
-				, static_cast<uint8_t>(alpha));
+		const int src_w = surf->w;
+		const int src_h = surf->h;
+
+		const float xratio = static_cast<float>(src_w) / static_cast<float>(w);
+		const float yratio = static_cast<float>(src_h) / static_cast<float>(h);
+		for(int ydst = 0; ydst != h; ++ydst) {
+			for(int xdst = 0; xdst != w; ++xdst) {
+				// Project dst pixel to a single corresponding src pixel by scale and simply take it
+				const int xsrc = std::floor(static_cast<float>(xdst) * xratio);
+				const int ysrc = std::floor(static_cast<float>(ydst) * yratio);
+				dst_pixels[ydst * dst->w + xdst] = src_pixels[ysrc * src_w + xsrc];
 			}
+		}
 
+		// In case format of the src surface is not the same as the default format for dst
+		// convert all pixels to the correct format
+		if(surf->format != dst->format) {
+			PLAIN_LOG << "Shite";
```

Comment:
Forgotten debug statement?

Reply:
@stevecotton Yeah, that's what I get for postponing stuff until late at night :(

By the way, how do I handle review commits? Do I add just new commits or maybe amend my commit and rebase force-push if it's a small thing? Is the pull request merged as is or with a squash?

Reply:
Whether it gets squashed or not depends on whether the person merging it remembers to change the setting in the drop-down menu, and on what they did to the last PR that they merged. This sometimes results in posting "oops ...." comments in the IRC/Discord channel.

So rebase and force-push is slightly preferred.